### DISCUSSION TITLE
Add support for Confluent Platform (with and without MDS)

### DISCUSSION THREAD

**Discussion by MSeal** [confluentinc/vscode#173] [Ideas]
### Goals

1. Enable CP clusters and their subresources to be usable in the same manner as CCloud
3. Handle most common auth and any feature gaps smoothly to eliminate mental burden for users

### Non-Goals

1. CP specific capabilities beyond those present in CCloud and local clusters
2. Data transfer between CP and other targets
4. Warpstream support (separate discussion?)

### Background

Many Confluent users have resources in their own managed cloud accounts via Confluent Platform and stream developers need to perform the same coding and testing paths they would for other environments. Additionally the abstractions are similar to the existing resource abstractions, requiring relatively low effort to convert into the existing UI and API calls. Intelij plugins and other non-IDE clients can simply connect to CP clusters to perform basic actions already, VS Code has no equivalent atm for these clusters.

### Proposal

In the `Resources` subpanel, if authenticated to a CP deployment the clusters from that platform should be visible. Clicking on those should expand the appropriate subpanel resources below (e.g. Topics, Schemas). Interaction below the resource panel should be identical to other target interactions, with perhaps minor differences in exposed capabilities (e.g. RBAC role differences).

TBD authentication pattern and UX for displaying auth alongside CCloud signin.

The Kafka and SR Kafka REST implementation under the API layer should reuse most if not all of the Direct Connect implementation for fulfilling API requests for CP.

This is a near term priority for the Confluent team to discuss deeper and implement.

**Comment by rhauch** [comment]
Tracking support for Confluent Platform (with and without MDS, and various authN mechanisms) using these milestones:
* https://github.com/confluentinc/vscode/milestone/4
* https://github.com/confluentinc/ide-sidecar/milestone/4
